File: android/app/src/main/java/com/colecoding/conduit/config/CommunityManager.kt
package com.colecoding.conduit.config

import android.content.Context
import android.content.SharedPreferences
import android.util.Log

/**
 * Manages the selected community URL for multi-tenant support
 */
object CommunityManager {
    private const val TAG = "CommunityManager"
    private const val PREF_NAME = "ConduitCommunityPrefs"
    private const val KEY_COMMUNITY_URL = "community_url"
    private const val KEY_COMMUNITY_NAME = "community_name"

    private fun getPrefs(context: Context): SharedPreferences {
        return context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    }

    fun getCommunityUrl(context: Context): String? {
        return getPrefs(context).getString(KEY_COMMUNITY_URL, null)
    }

    fun setCommunityUrl(context: Context, url: String) {
        Log.d(TAG, "Setting community URL: $url")
        getPrefs(context).edit().putString(KEY_COMMUNITY_URL, url).apply()
    }

    fun hasCommunityUrl(context: Context): Boolean {
        return getCommunityUrl(context) != null
    }

    fun clearCommunityUrl(context: Context) {
        Log.d(TAG, "Clearing community URL")
        getPrefs(context).edit().apply {
            remove(KEY_COMMUNITY_URL)
            remove(KEY_COMMUNITY_NAME)
            apply()
        }
    }

    fun getCommunityName(context: Context): String? {
        return getPrefs(context).getString(KEY_COMMUNITY_NAME, null)
    }

    fun setCommunityName(context: Context, name: String) {
        Log.d(TAG, "Setting community name: $name")
        getPrefs(context).edit().putString(KEY_COMMUNITY_NAME, name).apply()
    }
}

File: ios/Conduit/Conduit/Managers/CommunityManager.swift
import Foundation

/// Manages the selected community URL for multi-tenant support
class CommunityManager {
    static let shared = CommunityManager()

    private let communityURLKey = "conduit_community_url"
    private let communityNameKey = "conduit_community_name"
    private let communityDomainKey = "conduit_community_domain"

    private init() {}

    // MARK: - Community URL

    func getCommunityURL() -> URL? {
        guard let urlString = UserDefaults.standard.string(forKey: communityURLKey),
              let url = URL(string: urlString) else {
            return nil
        }
        return url
    }

    func setCommunityURL(_ url: URL) {
        UserDefaults.standard.set(url.absoluteString, forKey: communityURLKey)
    }

    func hasCommunityURL() -> Bool {
        return getCommunityURL() != nil
    }

    func clearCommunityURL() {
        UserDefaults.standard.removeObject(forKey: communityURLKey)
        UserDefaults.standard.removeObject(forKey: communityNameKey)
        UserDefaults.standard.removeObject(forKey: communityDomainKey)
    }

    // MARK: - Community Name (for display)

    func getCommunityName() -> String? {
        return UserDefaults.standard.string(forKey: communityNameKey)
    }

    func setCommunityName(_ name: String) {
        UserDefaults.standard.set(name, forKey: communityNameKey)
    }

    // MARK: - Community Domain (for authentication)

    func getCommunityDomain() -> String? {
        return UserDefaults.standard.string(forKey: communityDomainKey)
    }

    func setCommunityDomain(_ domain: String) {
        UserDefaults.standard.set(domain, forKey: communityDomainKey)
    }
}
